### PR TITLE
Feat/contract events

### PR DESCRIPTION
**Pull request by LordCheta**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (OnyekaIjeh) [inline .vscode/settings.json:1]: Should we be checking this is in?

File: contracts/FeeManager/FeeManager.sol [hunk 2]
```diff
@@ -88,6 +90,8 @@ contract FeeManager is IFeeManager, Ownable {
         if (!_rif.transfer(_pool, balance)) {
             revert TransferFailed(address(this), _pool, balance);
         }
+
+        emit DepositSuccessful(cost, partner);
```
Comment (OnyekaIjeh) [inline contracts/FeeManager/FeeManager.sol:94]: Same as prev comment

File: contracts/PartnerManager/PartnerManager.sol [hunk 3]
```diff
@@ -60,6 +60,8 @@ contract PartnerManager is IPartnerManager, Ownable {
         require(isPartner(partner), "PartnerManager: not a partner");
 
         _partnerConfigurations[partner] = partnerConfiguration;
+
+        emit PartnerConfigurationSet(partner, address(partnerConfiguration));
```
Comment (OnyekaIjeh) [inline contracts/PartnerManager/PartnerManager.sol:64]: Same as prev comment

File: contracts/PartnerManager/IPartnerManager.sol [hunk 4]
```diff
@@ -9,8 +9,21 @@ import "../PartnerConfiguration/IPartnerConfiguration.sol";
     @dev Keeps track of the whitelisted partners and its configurations.
 */
 interface IPartnerManager {
+    /**
+     * @notice event emitted when the configuration for a partner is set
+     * @param partner address for the partner
+     * @param configurationContract address of the configuration contract
+     */
+    event PartnerConfigurationSet(
```
Comment (OnyekaIjeh) [inline contracts/PartnerManager/IPartnerManager.sol:17]: I think `PartnerConfigurationChanged` is a better name, which also begs the question, should we allow updates if there are no changes?

File: contracts/Registrar/IBaseRegistrar.sol [hunk 5]
```diff
@@ -9,11 +9,24 @@ import "../FeeManager/IFeeManager.sol";
...
+     * @param duration the duration of the registration in years
+     */
     event NameRegistered(address indexed partner, uint256 duration);
 
+    /**
+     * @notice event emitted when a fee manager contract is set
+     * @param hostContract contract on which the fee manager is set
+     * @param feeManagerContract the address of the fee manager being set
+     */
+    event FeeManagerSet(address hostContract, address feeManagerContract);
```
Comment (OnyekaIjeh) [inline contracts/Registrar/IBaseRegistrar.sol:24]: See previous comment

File: contracts/Registrar/PartnerRegistrar.sol [hunk 6]
```diff
@@ -58,6 +58,8 @@ contract PartnerRegistrar is IBaseRegistrar, Ownable, IERC677TransferReceiver {
 
     function setFeeManager(IFeeManager feeManager) external onlyOwner {
         _feeManager = feeManager;
+
+        emit FeeManagerSet(address(this), address(_feeManager));
```
Comment (OnyekaIjeh) [inline contracts/Registrar/PartnerRegistrar.sol:62]: The event should be emitted before the state change.

File: contracts/Renewer/IBaseRenewer.sol [hunk 7]
```diff
@@ -18,5 +18,17 @@ interface IBaseRenewer {
...
+     * @param duration the duration of the renewal in years
+     */
     event NameRenewed(address indexed partner, uint256 duration);
+
+    /**
+     * @notice event emitted when a fee manager contract is set
+     * @param hostContract contract on which the fee manager is set
+     * @param feeManagerContract the address of the fee manager being set
+     */
+    event FeeManagerSet(address hostContract, address feeManagerContract);
```
Comment (OnyekaIjeh) [inline contracts/Renewer/IBaseRenewer.sol:33]: I think `FeeManagerChanged` is a better name

File: contracts/Renewer/PartnerRenewer.sol [hunk 8]
```diff
@@ -46,6 +46,8 @@ contract PartnerRenewer is IBaseRenewer, Ownable, IERC677TransferReceiver {
 
     function setFeeManager(IFeeManager feeManager) external onlyOwner {
         _feeManager = feeManager;
+
+        emit FeeManagerSet(address(this), address(_feeManager));
```
Comment (OnyekaIjeh) [inline contracts/Renewer/PartnerRenewer.sol:50]: Should come before state change

File: .vscode/settings.json (hunk 1 above)
Comment (LordCheta) [reply .vscode/settings.json:1]: Thanks for catching this!

File: contracts/Registrar/PartnerRegistrar.sol (hunk 6 above)
Comment (LordCheta) [reply contracts/Registrar/PartnerRegistrar.sol]: 👍🏾

File: contracts/Renewer/IBaseRenewer.sol (hunk 7 above)
Comment (LordCheta) [reply contracts/Renewer/IBaseRenewer.sol]: 👍🏾

File: contracts/Renewer/PartnerRenewer.sol (hunk 8 above)
Comment (LordCheta) [reply contracts/Renewer/PartnerRenewer.sol]: 👍🏾

File: test/FeeManager.spec.ts [hunk 9]
```diff
@@ -15,7 +15,9 @@ import { FeeManager__factory, RIF as RIFType } from 'typechain-types';
 import { PartnerManager } from '../typechain-types/contracts/PartnerManager/PartnerManager';
 import { PartnerConfiguration } from '../typechain-types/contracts/PartnerConfiguration/PartnerConfiguration';
 import { FakeContract, MockContract } from '@defi-wonderland/smock';
-import { Signer } from 'ethers';
+
+const DEPOSIT_SUCCESSFUL_EVENT = 'DepositSuccessful';
+const WITHDRAWAL_SUCCESSFUL_EVENT = 'WithdrawalSuccessful';
```
Comment (sergio-jara) [inline test/FeeManager.spec.ts:20]: utils file plis

File: test/FeeManager.spec.ts [hunk 10]
```diff
@@ -302,4 +304,90 @@ describe('Fee Manager', () => {
...
+          partnerOwnerAccount.address
+        );
+
+        await expect(
+          feeManager.connect(registrar).deposit(partner.address, depositAmount)
+        )
+          .to.emit(feeManager, DEPOSIT_SUCCESSFUL_EVENT)
+          .withArgs(depositAmount, partner.address);
+
+        // const partnerFee = depositAmount
```
Comment (sergio-jara) [inline test/FeeManager.spec.ts:343]: remove commented code

File: test/PartnerManager.spec.ts [hunk 11]
```diff
@@ -178,20 +181,72 @@ describe('partnerManager', () => {
             partner.address,
             partnerOwnerAccount.address
           )
-        ).to.not.be.reverted;
+        ).to.eventually.be.fulfilled;
```
Comment (sergio-jara) [inline test/PartnerManager.spec.ts:184]: nice

File: test/PartnerRegistrar.spec.ts [hunk 12]
```diff
@@ -382,4 +389,62 @@ describe('Registrar Checks', () => {
...
     }
   });
+
+  it('Should revert is the fee manager to be set is same as existing', async () => {
+    const { FeeManager, PartnerRegistrar } = await loadFixture(initialSetup);
+
+    await expect(
+      PartnerRegistrar.setFeeManager(FeeManager.address)
+    ).to.be.revertedWith(
+      'PartnerRegistrar: update param is same as param to be updated'
```
Comment (sergio-jara) [inline test/PartnerRegistrar.spec.ts:399]: utils file

**Review by sergio-jara — CHANGES_REQUESTED** [review]

**Review by sergio-jara — APPROVED** [review]

**Pull request merged**